src/resources/js/cart.js

```diff
@@ -0,0 +1,95 @@
+class Cart {
+    constructor(productObj) {
+        this.productObj = productObj;
+    }
+    viewProductsInCart() {
+        var userToken = localStorage.getItem('userToken');
+        $.ajax({
+            url: 'http://laravel-gernzy.test/graphql',
+            contentType: 'application/json',
+            type: 'POST',
+            context: this,
+            headers: {
+                Authorization: `Bearer ${userToken}`,
+            },
+            data: JSON.stringify({
+                query: `{
+                        me {
+                            cart {
+                                items {
+                                    product_id
+                                    quantity
+                                }
+                            }
+                        }
+                }`,
+            }),
+            success: function(result) {
+                let items = result.data.me.cart.items;
+                if (items && items.length > 0) {
+                    this.lookupProductsInCart(result.data.me.cart.items);
+                } else {
+                    $('.cart-products').html(`<div class="uk-alert-danger" uk-alert>
+                    <a class="uk-alert-close" uk-close></a>
+                    <p>No products in cart</p>
+                </div>`);
+
+                    // Disable checkout as there are no products in the cart
+                    $('#cart-checkout').addClass('uk-disabled');
+                }
+            },
+            error: function(XMLHttpRequest, textStatus, errorThrown) {
+                // console.log(XMLHttpRequest + textStatus + errorThrown);
+            },
+        });
+    }
+
+    async lookupProductsInCart(products) {
+        await Promise.all(
+            products.map(async product => {
+                const queriedProducts = await this.productObj.getProduct(product.product_id);
+                const mergedProduct = { ...queriedProducts, ...product };
+                return mergedProduct;
+            }),
+        )
+            .then(re => {
+                this.populateUIWithProducts(re);
+            })
+            .catch(error => {
+                console.log(error);
+            });
+    }
+
+    populateUIWithProducts(products) {
+        let container = $('<div class="uk-flex uk-flex-wrap uk-flex-wrap-around ">');
+
+        products.forEach(function(product) {
+            container.append(`
+            <div>
```

Comment:
@AlexeiaW we can pull templates out into a separate set of files, each file can just export the template as a function so you can continue to do string replacement. Something like

templates/products.js
templates/checkout.js

Reply:
Ok I implemented templating and will do the same for future components @luke-siedle